File: Backend/Sadef.Domain/PropertyEntity/Property.cs
using Sadef.Common.Domain;
using Sadef.Domain.Constants;

namespace Sadef.Domain.PropertyEntity
{
    public class Property : AggregateRootBase
    {
        // Static fields (non-translatable) - kept in main table
        public required decimal Price { get; set; }
        public required PropertyType PropertyType { get; set; }
        public UnitCategory? UnitCategory { get; set; }
        public required City? City { get; set; }
        public required string Location { get; set; }
        public required double AreaSize { get; set; }
        public int? Bedrooms { get; set; }
        public int? Bathrooms { get; set; }
        public ICollection<PropertyImage>? Images { get; set; }
        public PropertyStatus Status { get; set; } = PropertyStatus.Pending;
        public DateTime? ExpiryDate { get; set; }
        public ICollection<PropertyVideo>? Videos { get; set; }
        public decimal? ProjectedResaleValue { get; set; }
        public decimal? ExpectedAnnualRent { get; set; }
        public double? Latitude { get; set; }
        public double? Longitude { get; set; }
        public string? WhatsAppNumber { get; set; }
        public DateTime? ExpectedDeliveryDate { get; set; }
        public bool IsInvestorOnly { get; set; } = false;
        public List<FeatureList>? Features { get; set; }
        public int? TotalFloors { get; set; }
        public bool? IsActive { get; set; } = true;
        public ContentLanguage ContentLanguage { get; set; } = ContentLanguage.English;
        public ICollection<PropertyTranslation> Translations { get; set; } = new List<PropertyTranslation>();
    }
}

File: Frontend/src/types/property.ts
export interface PropertyFormData {
    // Basic Info
    title: string;
    propertyType?: number;
    unitCategory?: number;
    price: number;
    city: string;
    location: string;
    areaSize: number;
    bedrooms?: number;
    bathrooms?: number;
    totalFloors?: number;
    unitName?: string;
    isInvestorOnly?: boolean;

    // Property Details
    description: string;
    features?: string[];
    projectedResaleValue?: number;
    expectedAnnualRent?: number;
    warrantyInfo?: string;
    expectedDeliveryDate?: string;
    status?: number;
    expiryDate?: string;

    // Property Media
    images?: File[];
    videos?: File[];

    // Location/Map
    latitude?: number;
    longitude?: number;

    // Contact & Publishing
    whatsAppNumber?: string;
}

export type CreatePropertyData = {
    id: number;
    title: string;
    description: string;
    price: number;
    propertyType?: number;
    unitCategory?: number;
    city: string;
    location: string;
    areaSize: number;
    bedrooms?: number;
    bathrooms?: number;
    totalFloors?: number;
    images?: string[];
    status?: number;
    videos?: string[];
    unitName?: string;
    projectedResaleValue?: number;
    expectedAnnualRent?: number;
    warrantyInfo?: string;
    latitude?: number;
    longitude?: number;
    whatsAppNumber?: string;
    expectedDeliveryDate?: string;
    expiryDate?: string;
    isInvestorOnly?: boolean;
    features?: number[];
    createdAt?: string;
    updatedAt?: string;
    imageBase64Strings?: string[];
    videoBase64Strings?: string[];
}

export type CreatePropertyResponse = {
    data: CreatePropertyData;
    message: string;
    succeeded: boolean;
}

export type PropertyItem = {
    id: number;
    title: string;
    description: string;
    price: number;
    propertyType: number;
    unitCategory: number;
    city: string;
    location: string;
    areaSize: number;
    bedrooms: number;
    bathrooms: number;
    imageBase64Strings: string[];
    status: number;
    expiryDate: string | null;
    isExpired: boolean;
    videoUrls: string[] | null;
    unitName: string;
    projectedResaleValue: number;
    expectedAnnualRent: number;
    warrantyInfo: string;
    latitude: number;
    longitude: number;
    whatsAppNumber: string;
    expectedDeliveryDate: string;
    isInvestorOnly: boolean;
    features: number[];
    isActive: boolean;
}

type PropertyData = {
    items: PropertyItem[];
};

export type GetProperties = {
    succeeded: boolean,
    message: string | null,
    validationResultModel: string | null,
    data: PropertyData,
    totalCount: number,
    pageNumber: number,
    pageSize: number,
    extra: string | null,
}